### ISSUE TITLE
This API is a work in progress and will likely be changing frequently. 

### ISSUE THREAD

**Issue by piascikj** [imdone/tensorflow#85]
/
/ Given initial gradients 'grad_inputs' (which represent the symbolic partial
/ derivatives of some loss function 'L' w.r.t 'outputs'), adds gradient nodes
/ to the graph associated with 'scope', which compute (and return in
/ 'grad_outputs') the symbolic partial derivatives of 'L' w.r.t 'inputs'.

----
*opened via [imdone.io](https://imdone.io) from a code comment on [23441bf71dbd0a10bcace160165ce01bf6deac9a](https://github.com/imdone/tensorflow/commit/23441bf71dbd0a10bcace160165ce01bf6deac9a) by A. Unique TensorFlower gardener@tensorflow.org*

----

https://github.com/imdone/tensorflow/blob/8cb6e535f6ea14b380aa86d425169adad682cb8c/tensorflow/cc/framework/gradients.h#L24-L34
